packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -960,6 +1037,12 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     workInProgress: Fiber,
     renderExpirationTime: ExpirationTime,
   ): Fiber | null {
+    // The effect list is no longer valid.
+    workInProgress.nextEffect = null;
+    workInProgress.firstEffect = null;
+    workInProgress.lastEffect = null;
+    workInProgress.thrownValue = null;
```

Comment:
Why did this move? We should be careful not to add things to things that happen in the hottest paths unless they *always* need to happen. If they sometimes need to happen, it should only happen in those paths.